modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/CurrencySnapshotConsensusFunctions.scala

```diff
@@ -36,30 +31,17 @@ abstract class CurrencySnapshotConsensusFunctions[F[_]: Async: SecurityProvider:
 
 object CurrencySnapshotConsensusFunctions {
 
-  def make[F[_]: Async: KryoSerializer: Hasher: SecurityProvider: L0NodeContext](
-    stateChannelSnapshotService: StateChannelSnapshotService[F],
+  def make[F[_]: Async: KryoSerializer: Hasher: SecurityProvider](
     collateral: Amount,
     rewards: Option[Rewards[F, CurrencySnapshotStateProof, CurrencyIncrementalSnapshot, CurrencySnapshotEvent]],
     currencySnapshotCreator: CurrencySnapshotCreator[F],
-    currencySnapshotValidator: CurrencySnapshotValidator[F],
-    gossip: Gossip[F],
-    maybeDataApplication: Option[BaseDataApplicationL0Service[F]]
+    currencySnapshotValidator: CurrencySnapshotValidator[F]
   ): CurrencySnapshotConsensusFunctions[F] = new CurrencySnapshotConsensusFunctions[F] {
 
     def getRequiredCollateral: Amount = collateral
 
     def getBalances(context: CurrencySnapshotContext): SortedMap[Address, Balance] = context.snapshotInfo.balances
 
-    def consumeSignedMajorityArtifact(
```

Comment:
Not needed on the interface anymore. Consensus steps are customizable right now, so the consume logic can be implemented (and is in this PR) as the effect on the last step of consensus.